core/lib/tasks/payment_method.rake

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+namespace :payment_method do
+  desc "Deactivates old payment methods and fixes ActiveRecord::SubclassNotFound error, "\
+  "which happens after switching Payment Service Provider."
+  task deactivate_unsupported_payment_methods: :environment do
+    Spree::PaymentMethod.pluck(:id, :type).select do |id, type|
+      begin
+        ActiveSupport::Dependencies.constantize(type)
+      rescue NameError
+        fix_payment_method_record(id, type)
+      end
+    end
+  end
+
+  def fix_payment_method_record(id, previous_type)
+    connection = ActiveRecord::Base.connection
+    false_value = connection.quoted_false
+    connection.exec_update(<<-SQL
+      UPDATE spree_payment_methods
+      SET type='#{Spree::PaymentMethod.name}', previous_type='#{previous_type}',
+      active=#{false_value}, available_to_users=#{false_value}, available_to_admin=#{false_value}
```

Comment:
Not mandatory, but I'd move all these attributes to separate lines. This will allow us to add/change/remove individual lines in the future without impacting git history for all of them.

Reply:
Fixed it, actually it was a bit ugly before.